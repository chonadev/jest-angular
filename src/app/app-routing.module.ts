import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CounterComponent } from './basic/counter/counter.component';
import { RouterModule, Routes } from '@angular/router';
import { CharizardComponent } from './basic/charizard/charizard.component';

const routes: Routes = [
  { 
    path: "basic/counter",
    component: CounterComponent
  },
  {
    path: "basic/charizard",
    component: CharizardComponent
  },
  {
    path: "**",
    redirectTo: "basic/charizard"
  }
]

@NgModule({
  declarations: [],
  imports: [
    RouterModule.forRoot(routes),
    CommonModule
  ],
  exports: [
    RouterModule
  ]
})
export class AppRoutingModule { }
