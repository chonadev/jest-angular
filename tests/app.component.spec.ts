import { TestBed } from '@angular/core/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  // test(`should have as title 'jest-angular'`, () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   const component = fixture.componentInstance;
  //   expect(component.title).toEqual('jest-angular');
  // });

  // test('should render title', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   const component = fixture.componentInstance;
  //   fixture.detectChanges();
  //   const compiled = fixture.nativeElement as HTMLElement;
  //   const h1 = compiled.querySelector('h1');
  //   expect(h1?.textContent).toContain(component.title);
  // });

  // test('debe hacer match con el snapshot', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   fixture.detectChanges();
  //   const component = fixture.componentInstance;

  //   expect( component ).toMatchSnapshot();

  // });
});
