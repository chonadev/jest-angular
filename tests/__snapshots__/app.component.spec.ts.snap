// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent debe hacer match con el snapshot 1`] = `
AppComponent {
  "__ngContext__": 6,
  "title": "jest-angular",
}
`;
